Sources/XCTest/XCTestCase.swift

```diff
@@ -93,6 +107,24 @@ extension XCTestCase {
 
                 testCase.tearDown()
 
+                // It is an API violation to create expectations but not wait
```

Comment:
This entire block feels like something that could be extracted into a private helper. The `invokeTests` function is getting a bit unwieldy!